spec/lib/services/sku-pack-service-spec.js

```diff
@@ -183,18 +187,154 @@ describe("SKU Pack Service", function() {
               "updatedAt": "2016-04-07T12:25:18.529Z",
               "id": "570651ae87b3579d76508d26"
           };
-            waterline.skus.create.resolves(sku);
-            waterline.skus.findOne.resolves();
+          waterline.skus.create.resolves(sku);
+          waterline.skus.findOne.resolves();
+          skuService.regenerateSkus.resolves();
+          return skuService.postSku(sku).then(function(val){
+              expect(waterline.skus.create).to.have.been.called;
+              expect(val.rules).equal(sku.rules);
+              expect(val.discoveryGraphName).equal(sku.discoveryGraphName);
+              expect(val.username).equal(sku.username);
+              expect(val.password).equal(sku.password);
+              expect(val.hostname).equal(sku.hostname);
+          });
+        });
+
+        it('should return an error Duplicate name found', function() {
+            var sku =
+            {"name": "my test sku"};
+            waterline.skus.findOne.resolves({});
+            return skuService.postSku(sku).catch(function(err){
```

Comment:
Same as my last comment, need some special code to prevent the `postSku` from being resolved.